torch/csrc/autograd/python_variable.cpp

```diff
@@ -224,12 +224,7 @@ PyObject *THPVariable_get_grad(THPVariable *self)
 {
   auto& var = *self->cdata;
   if (!var.grad) {
-#ifdef WITH_CUDA
-    THCPAutoGPU __guard(var.data->getDevice());
-#endif
-    auto grad = var.data->newTensor();
-    grad->resizeAs(*var.data).zero();
-    var.grad = std::make_shared<Variable>(std::move(grad), 0, 1);
+    return Py_None;
```

Comment:
When gradients can be sparse, there is unfortunately no good default value for zero gradients (the gradient could be dense or sparse, and even for sparse gradients we don't know how many dimensions are actually sparse). I think the most elegant way to handle it is to leave it to `None` by default, though I understand this is a small breaking change. Let me know if you see a better way.